### PR TITLE
feat: support actions message in shell adapter

### PR DESCRIPTION
**Pull request by tcnguyen**
Sometimes it is useful to handle actions message in shell for development purpose (no need to setup webchat...)

==> bot proposes choices for action
screen shot 2018-04-24 at 15 05 34

### CODE REVIEW

File: packages/botfuel-dialog/src/adapters/shell-adapter.js [hunk 1]
```diff
@@ -50,10 +52,47 @@ class ShellAdapter extends Adapter {
...
+   * @param {Object} actionsMessage actionMessage
+   * @returns {Promise.<void>}
+   */
+  async sendActionsMessage(actionsMessage) {
+    const actions = actionsMessage.payload.value;
+    let str = 'Please select:\n';
+    actions.forEach((action, index) => {
+      str += `${index + 1}: (${action.type}) ${action.text}\n`;
+    });
+
```
Comment (windkomo) [inline packages/botfuel-dialog/src/adapters/shell-adapter.js]: "Pure" way of doing this (not really important, can be useful if you decide to put this in a function outside `sendActionsMessage`):

```javascript
const choices = actions
    .map((action, index) => `${ index + 1 }: (${action.type}) ${action.text}\n`);

const question = `Please select:\n${choices}'
```

File: packages/botfuel-dialog/src/adapters/shell-adapter.js [hunk 2]
```diff
@@ -50,10 +52,47 @@ class ShellAdapter extends Adapter {
...
+    this.rl.question(chalk.hex(this.botTextColor)(str), async (answer) => {
+      const id = parseInt(answer, 10);
+      if (id >= 1 && id <= actions.length) {
+        const selected = actions[id - 1];
+        // postback
```
Comment (windkomo) [inline packages/botfuel-dialog/src/adapters/shell-adapter.js]: This comment is not needed in my opinion 😄

File: packages/botfuel-dialog/src/adapters/shell-adapter.js [hunk 3]
```diff
@@ -50,10 +52,47 @@ class ShellAdapter extends Adapter {
...
+        if (selected.type === 'postback') {
+          const userMessage = new PostbackMessage(
+            selected.value.dialog,
+            selected.value.entities,
+          ).toJson(this.userId);
+          await this.handleMessage(userMessage);
+          // link
```
Comment (windkomo) [inline packages/botfuel-dialog/src/adapters/shell-adapter.js]: Same for this one!

File: packages/botfuel-dialog/src/adapters/shell-adapter.js [hunk 4]
```diff
@@ -50,10 +52,47 @@ class ShellAdapter extends Adapter {
...
+   * @returns {Promise.<void>}
+   */
+  async sendActionsMessage(actionsMessage) {
+    const actions = actionsMessage.payload.value;
+    let str = 'Please select:\n';
+    actions.forEach((action, index) => {
+      str += `${index + 1}: (${action.type}) ${action.text}\n`;
+    });
+
+    this.rl.question(chalk.hex(this.botTextColor)(str), async (answer) => {
```
Comment (michelbl) [inline packages/botfuel-dialog/src/adapters/shell-adapter.js]: Are you sure the answer does not trigger an event `line`?
(an event listener is registered `this.rl.on('line', async (input) => {...`)

Comment (tcnguyen) [reply packages/botfuel-dialog/src/adapters/shell-adapter.js]: I tested and it worked fine as in the image in the PR description

**Pull request merged**